server/src/main/java/org/opensearch/search/backpressure/settings/SearchTaskSettings.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.backpressure.settings;
+
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.monitor.jvm.JvmStats;
+
+/**
+ * Defines the settings related to the cancellation of SearchTasks.
+ *
+ * @opensearch.internal
+ */
+
+public class SearchTaskSettings {
+    private static final long HEAP_SIZE_BYTES = JvmStats.jvmStats().getMem().getHeapMax().getBytes();
```

Comment:
Please by aware that get max heap *may be undefined* as per `java.lang.management.MemoryUsage`:

```
/**
     * Returns the maximum amount of memory in bytes that can be
     * used for memory management.  This method returns {@code -1}
     * if the maximum memory size is undefined.
     *
```

So the HEAP_SIZE_BYTES would be set to `0` by `JvmStats.jvmStats()` implementation. We should account for that or risking to cancel everything (getTotalHeapBytesThreshold() would return 0)

Reply:
@PritLadani ^^ please

Reply:
Missed addressing this. Can you please suggest a workaround for this as well? We cannot take the risk of cancelling all the tasks, and we cannot assume any value for `HEAP_SIZE_BYTES` as well.

Reply:
I think the workaround is not to cancel tasks (at least by heap threshold) since we cannot estimate the heap usage.

Reply:
Makes sense. I will address this in the next commit.